staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -780,6 +781,23 @@ type costInfo struct {
 	// definition. A single totalCost is allocated for each validation call and passed through the stack as the
 	// custom resource definition's OpenAPIv3 schema is recursively validated.
 	TotalCost *totalCost
+	// Schema is a lazily populated structural schema for this costInfo node.
+	Schema *structuralschema.Structural
+	// DeclType is a lazily populated CEL declaration representation of Schema of this costInfo node.
+	DeclType *model.DeclType
```

Comment:
it's a touch weird to put this inside a costInfo type... is there a clearer name for the type if it's going to expand to be used this way?